### ISSUE TITLE
[a11y]: File uploader focus handling after file delete

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11204]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--drag-and-drop-upload-container-example-application

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS and keyboard

### Description

After a file is deleted from the upload list, the focus does not go anywhere. (I'll mention that the delete action does not have a confirmation, but set that aside as a design decision.) 
https://ibm.box.com/s/scxudnpq13sqvnv0jbqxamlcyp31r8kf

Prior CAG discussions on focus handling are relevant to this problem. Because the item with focus is being deleted, where should the focus go?
In prior discussions, the thought was to go the former item in the tab order. Until the delete buttons are distinguishable programmatically (#11198) it definitely makes more sense to put it on the File add button. And even once that is resolved, I think it may be a more consistent experience to always go to the File upload button regardless of which file has been delete 

We'd previously discussed the notion of a group of controls. Perhaps an interactive parent should be part of the algorithm. This same algorithm could work for table row deletions, menu item deletions and the file uploaded group of files situation.

### WCAG 2.1 Violation

https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--drag-and-drop-upload-container-example-application

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-fileuploader--drag-and-drop-upload-container-example-application
2. Upload a few files
3. Delete a file
  - the focus will not be anywhere visually

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
